plugin/pkg/scheduler/scheduler.go

```diff
@@ -258,6 +309,11 @@ func (sched *Scheduler) scheduleOne() {
 	suggestedHost, err := sched.schedule(pod)
 	metrics.SchedulingAlgorithmLatency.Observe(metrics.SinceInMicroseconds(start))
 	if err != nil {
+		// schedule() may have failed because the pod would not fit on any host, so we try to
+		// preempt, with the expectation that the next time the pod is tried for scheduling it
+		// will fit due to the preemption. It is also possible that a different pod will schedule
+		// into the resources that were preempted, but this is harmless.
+		sched.preempt(pod, err)
```

Comment:
calling this without checking if the error is a FitError could make for very noisy logs when the feature gate is disabled

Reply:
preempt checks the error and if it's not FitError returns.

Reply:
not before logging if the feature is disabled

Reply:
Added a check.